### ISSUE TITLE
Extension search is broken

### ISSUE THREAD

**Issue by eamodio** [microsoft/vscode#85178]
Issue Type: <b>Bug</b>

Steps:

1. In the extensions viewlet, right click and hide the "Recommended" view.
2. Type in search box - 🐛  get no results

Seemed to have started with yesterday's insiders. If you type in the extensions search box, you get no results and the title says `Disabled`. See below:
![extensions](https://user-images.githubusercontent.com/641685/69216292-dbfc2e00-0b39-11ea-8bd8-5e33f9631388.gif)

VS Code version: Code - Insiders 1.41.0-insider (4934a6f487b9ed6b99b900ae0710c7b452792e20, 2019-11-20T05:25:39.619Z)
OS version: Windows_NT x64 10.0.19025

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-6700K CPU @ 4.00GHz (8 x 4008)|
|GPU Status|2d_canvas: enabled<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>metal: disabled_off<br>multiple_raster_threads: enabled_on<br>oop_rasterization: disabled_off<br>protected_video_decode: unavailable_off<br>rasterization: enabled<br>skia_renderer: disabled_off<br>surface_control: disabled_off<br>surface_synchronization: enabled_on<br>video_decode: enabled<br>viz_display_compositor: enabled_on<br>viz_hit_test_surface_layer: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|31.93GB (17.95GB free)|
|Process Argv||
|Screen Reader|no|
|VM|0%|
</details>

**Comment by egamma** [comment]
Cannot reproduce using the latest insiders. Can you still reproduce this?

![image](https://user-images.githubusercontent.com/172399/69236614-2dd1a200-0b94-11ea-8a09-4f7be7f313d9.png)

**Comment by eamodio** [comment]
Yeah, even with the 2nd insiders build today.

**Comment by RMacfarlane** [comment]
I see the same thing
<img width="1102" alt="Screen Shot 2019-11-20 at 9 01 50 AM" src="https://user-images.githubusercontent.com/3672607/69260240-6f525500-0b74-11ea-8178-20b93c4eaf64.png">

**Comment by deepak1556** [comment]
I tested both the builds on my windows machine even with a fresh user data dir, couldn't repro.

What headers are sent for the cancelled request ?

Can you also try launching vscode with `--disable-features="OutOfBlinkCors"` and check ?

**Comment by RMacfarlane** [comment]
Headers are

* Accept: application/json;api-version=3.0-preview.1
* Content-Type: application/json
* Origin: file://
* Sec-Fetch-Mode: cors
* User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_14_6) AppleWebKit/537.36 (KHTML, like Gecko) Code-Insiders/1.41.0-insider Chrome/76.0.3809.146 Electron/6.1.4 Safari/537.36
* X-Market-Client-Id: VSCode 1.41.0-insider
* X-Market-User-Id: 7e8a9fb5-fe45-4bed-92c4-761a115ab42a

Though I'm now thinking the cancelled request might be unrelated? I also see an initial cancelled request on stable, followed by a successful request.
![Screen Shot 2019-11-20 at 4 12 57 PM](https://user-images.githubusercontent.com/3672607/69289245-bf4f0d00-0bb0-11ea-9dc0-feb1dd4f6abb.png)

 If I continue to type in the input box on insiders, no further requests get made.

We seem to ignore the `disable-features` flag, I get `Option 'disable-features' is unknown. Ignoring.` when trying to use it

**Comment by deepak1556** [comment]
> We seem to ignore the disable-features flag

Ah yeah needs to be fixed in https://github.com/microsoft/vscode/issues/84564

Can you verify this from out of sources build, in that case make the following change.

```
diff --git a/src/main.js b/src/main.js
index d5f4c907fe..c800b74d30 100644
--- a/src/main.js
+++ b/src/main.js
@@ -143,6 +143,8 @@ function configureCommandlineSwitchesSync(cliArgs) {
                        return; // unsupported argv key
                }

+               app.commandLine.appendSwitch('disable-features', 'OutOfBlinkCors');
+
                const argvValue = argvConfig[argvKey];
                if (argvValue === true || argvValue === 'true') {
                        if (argvKey === 'disable-hardware-acceleration') {
```

**Comment by sandy081** [comment]
I am also not able to repro it. 

@eamodio @RMacfarlane Can you please try reproducing it from sources. 

Note you have to enable Marketplace for that. I will ping you directly how to enable it. 
It would be helpful to nail it down if you can repro from out of sources.

**Comment by sandy081** [comment]
Re request cancellation, we cancel it if there is a new request.

**Comment by sandy081** [comment]
@alexr00 was also seeing this weird behaviour. It disappeared after updating insiders. 

@eamodio Can you please try with latest insiders?

**Comment by RMacfarlane** [comment]
Ok, I have repro steps now. I still see this in the latest insiders, and it's related to what views are shown.

1. In the extensions viewlet, right click and hide the "Recommended" view.
2. Type in search box - get no results

This causes the error

```
 ERR view descriptor extensions.disabledExtensionList not found: Error: view descriptor extensions.disabledExtensionList not found
    at PersistentContributableViewsModel.find (file:///Users/ramacfar/dev/vscode/out/vs/workbench/browser/parts/views/views.js:280:19)
    at PersistentContributableViewsModel.setCollapsed (file:///Users/ramacfar/dev/vscode/out/vs/workbench/browser/parts/views/views.js:224:59)
    at file:///Users/ramacfar/dev/vscode/out/vs/workbench/browser/parts/views/viewsViewlet.js:161:37
    at Emitter.fire (file:///Users/ramacfar/dev/vscode/out/vs/base/common/event.js:458:38)
    at file:///Users/ramacfar/dev/vscode/out/vs/base/common/event.js:55:106
    at Emitter.fire (file:///Users/ramacfar/dev/vscode/out/vs/base/common/event.js:458:38)
    at file:///Users/ramacfar/dev/vscode/out/vs/base/common/event.js:43:93
    at Emitter.fire (file:///Users/ramacfar/dev/vscode/out/vs/base/common/event.js:458:38)
    at DisabledExtensionsView.set headerVisible [as headerVisible] (file:///Users/ramacfar/dev/vscode/out/vs/base/browser/ui/splitview/paneview.js:106:31)
    at ExtensionsViewlet.updateViewHeaders (file:///Users/ramacfar/dev/vscode/out/vs/workbench/browser/parts/views/paneViewlet.js:317:54)
    at ExtensionsViewlet.removePanes (file:///Users/ramacfar/dev/vscode/out/vs/workbench/browser/parts/views/paneViewlet.js:278:18)
    at ExtensionsViewlet.onDidRemoveViews (file:///Users/ramacfar/dev/vscode/out/vs/workbench/browser/parts/views/viewsViewlet.js:183:18)
    at file:///Users/ramacfar/dev/vscode/out/vs/workbench/browser/parts/views/viewsViewlet.js:42:57
    at Emitter.fire (file:///Users/ramacfar/dev/vscode/out/vs/base/common/event.js:458:38)
    at PersistentContributableViewsModel.onDidChangeViewDescriptors (file:///Users/ramacfar/dev/vscode/out/vs/workbench/browser/parts/views/views.js:341:35)
    at file:///Users/ramacfar/dev/vscode/out/vs/workbench/browser/parts/views/views.js:175:91
    at Emitter.fire (file:///Users/ramacfar/dev/vscode/out/vs/base/common/event.js:458:38)
    at ViewDescriptorCollection.onContextChanged (file:///Users/ramacfar/dev/vscode/out/vs/workbench/browser/parts/views/views.js:148:35)
    at Emitter.fire (file:///Users/ramacfar/dev/vscode/out/vs/base/common/event.js:461:41)
    at file:///Users/ramacfar/dev/vscode/out/vs/base/common/event.js:55:106
    at PauseableEmitter.fire (file:///Users/ramacfar/dev/vscode/out/vs/base/common/event.js:458:38)
    at PauseableEmitter.fire (file:///Users/ramacfar/dev/vscode/out/vs/base/common/event.js:519:27)
    at ContextKeyService.setContext (file:///Users/ramacfar/dev/vscode/out/vs/platform/contextkey/browser/contextKeyService.js:247:42)
    at ContextKey.set (file:///Users/ramacfar/dev/vscode/out/vs/platform/contextkey/browser/contextKeyService.js:143:27)
    at ExtensionsViewlet.doSearch (file:///Users/ramacfar/dev/vscode/out/vs/workbench/contrib/extensions/browser/extensionsViewlet.js:422:41)
    at file:///Users/ramacfar/dev/vscode/out/vs/workbench/contrib/extensions/browser/extensionsViewlet.js:396:51
    at file:///Users/ramacfar/dev/vscode/out/vs/base/common/async.js:174:28
```

and this error is causing the token that gets passed through to the request service to get cancelled.

@sbatten Could this be related to any layout refactoring?

**Comment by RMacfarlane** [comment]
@sbatten I think the problem is here https://github.com/microsoft/vscode/blob/1ca17cd262c53f632a56ccadf92a3d38ee49cfd0/src/vs/workbench/browser/parts/views/viewsViewlet.ts#L242

The concept of "index" here is confusing, it appears to be the index of the view taking into account both visible and hidden views. So the "Disabled" view has index 2 even though it is displayed at index 1.

**Comment by RMacfarlane** [comment]
@sandy081, @sbatten's change was only a rename, after reverting to before 744b5161318a2ebcbd2f0fe0d3124697487907b1 the search works as expected

**Comment by sandy081** [comment]
@RMacfarlane Thanks for the steps and I can repro it too. I guess this is caused due to the fix I did here - #79519. I am actually emitting events with wrong visible indices. Hence views are getting shown and hidden like crazy. 

Working on the fix.

**Comment by sandy081** [comment]
Confirmed that this caused by my fix in #79519. 
109fab8 fixes this. Sorry for this.
